### PR TITLE
MessageBroker tests

### PR DESCRIPTION
**Pull request by Hroy02**

### CODE REVIEW

File: src/test/java/bgu/spl/mics/MessageBrokerTest.java [hunk 1]
```diff
@@ -1,19 +1,105 @@
...
+    public void testSingleton() {
+        MessageBroker otherInstance = MessageBrokerImpl.getInstance();
+        assertTrue(otherInstance == m, "different instances of singleton MessageBroker exist!");
+    }
+
+    @Test
+    public void testEventSubscribe() {
+        m.sendEvent(new IntEvent(0));
+        m.sendEvent(new IntEvent(1));
+        safeAwaitMessage(true);
```
Comment (IdanTo1) [inline src/test/java/bgu/spl/mics/MessageBrokerTest.java:47]: move deceleration above this method call

File: src/test/java/bgu/spl/mics/MessageBrokerTest.java [hunk 2]
```diff
@@ -1,19 +1,105 @@
...
+        assertTrue(s1.getNum() == 0, "Subscriber 1 didn't get the first called event");
+        assertTrue(s2.getNum() == 1, "Subscriber 2 didn't get the second called event");
+        m.sendEvent(new IntEvent(2));
+        safeAwaitMessage( false);
+        assertTrue(s1.getNum() == 2, "Subscriber 1 didn't appear 3rd in a 2 object round-robin");
```
Comment (IdanTo1) [inline src/test/java/bgu/spl/mics/MessageBrokerTest.java:52]: I would add a check for s2. to see it is still 1

**Review by IdanTo1 — DISMISSED** [review]

File: src/test/java/bgu/spl/mics/MessageBrokerTest.java (hunk 1 above)
Comment (Hroy02) [reply src/test/java/bgu/spl/mics/MessageBrokerTest.java:47]: Do you mean the declaration of safeAwaitMessage?

Comment (IdanTo1) [reply src/test/java/bgu/spl/mics/MessageBrokerTest.java:47]: Yes. We always prefer a method definition appear before it is used

File: src/test/java/bgu/spl/mics/MessageBrokerTest.java (hunk 2 above)
Comment (Hroy02) [reply src/test/java/bgu/spl/mics/MessageBrokerTest.java:52]: Done

File: src/test/java/bgu/spl/mics/MessageBrokerTest.java (hunk 1 above)
Comment (Hroy02) [reply src/test/java/bgu/spl/mics/MessageBrokerTest.java:47]: Done

**Review by IdanTo1 — APPROVED** [review]
great Job
next time please don't mark things as resolved (:

**Pull request merged**